### PR TITLE
feat: Add quantile statistic

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/block_transforms.py [hunk 1]
```diff
@@ -105,6 +106,37 @@ def indicate_duplicates(
...
+    qs: Sequence[float],
+    grouping_column_ids: Sequence[str] = (),
+) -> blocks.Block:
+    # TODO: handle windowing and more interpolation methods
+    window = core.WindowSpec(
+        grouping_keys=tuple(grouping_column_ids),
+    )
+    quantile_cols = []
+    labels = []
+    for col in columns:
```
Comment (chelsea-lin) [inline bigframes/core/block_transforms.py:121]: Throws an exception for dataframe if column_name is larger than 30?
https://github.com/googleapis/python-bigquery-dataframes/blob/0e24036fe33c60cef08ba78e7ce69319338b6b03/bigframes/core/blocks.py#L1151

**Review by chelsea-lin — CHANGES_REQUESTED** [review]

File: bigframes/core/block_transforms.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/core/block_transforms.py:123]: added limit

File: bigframes/series.py [hunk 2]
```diff
@@ -968,10 +968,19 @@ def mean(self) -> float:
...
         if exact:
-            raise NotImplementedError(
-                f"Only approximate median is supported. {constants.FEEDBACK_LINK}"
-            )
-        return typing.cast(float, self._apply_aggregation(agg_ops.median_op))
+            return self.quantile(0.5)
+        else:
+            return typing.cast(float, self._apply_aggregation(agg_ops.median_op))
+
+    def quantile(self, q: float) -> Union[Series, float]:
```
[the remark this answers is not in the document]
Comment (TrevorBergeron) [reply bigframes/series.py]: added

File: bigframes/dataframe.py [hunk 3]
```diff
@@ -2012,6 +2012,23 @@ def median(
         block = frame._block.aggregate_all_and_stack(agg_ops.median_op)
         return bigframes.series.Series(block.select_column("values"))
 
+    def quantile(self, q):
```
[the remark this answers is not in the document]
Comment (TrevorBergeron) [reply bigframes/dataframe.py]: added

[the remark this answers is not in the document]
Comment (chelsea-lin) [reply bigframes/dataframe.py]: the doc also need to be updated.

**Review by chelsea-lin — APPROVED** [review]

**Pull request merged**